File: BackEND/Monolith/Monolith/Entities/Opportunities/Opportunity.cs
namespace Monolith.Entities;

public class Opportunity
{
    public long Id { get; set; }
    public long CompanyId { get; set; }
    public long CreatedByUserId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string ShortDescription { get; set; } = string.Empty;
    public string FullDescription { get; set; } = string.Empty;
    public OpportunityType OppType { get; set; }
    public WorkFormat Format { get; set; }
    public OpportunityStatus Status { get; set; } = OpportunityStatus.Draft;
    public long? CityId { get; set; }
    public long? LocationId { get; set; }
    public decimal? SalaryFrom { get; set; }
    public decimal? SalaryTo { get; set; }
    public string? CurrencyCode { get; set; }
    public DateTimeOffset PublishAt { get; set; }
    public DateTimeOffset? ApplicationDeadline { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }
    public Company Company { get; set; } = null!;
    public City? City { get; set; }
    public Location? Location { get; set; }
    public List<OpportunityTag> OpportunityTags { get; set; } = new();
}

File: FrontEND/src/types/opportunity.ts
import type { OpportunityType as MockOpportunityType } from '../data/mockData'

export type OpportunityType = MockOpportunityType

export type Opportunity = {
  id: number
  entityType?: 'vacancy' | 'opportunity'
  title: string
  type: OpportunityType
  status: number
  compensation: string
  company: string
  location: string
  workFormat: string
  date: string
  description: string
  tags: string[]
  verified: boolean
  favoriteCompany?: boolean
  latitude?: number | null
  longitude?: number | null
}

export type OpportunityDetail = Opportunity & {
  shortDescription: string
  fullDescription: string
  publishAt: string | null
  applicationDeadline: string | null
  isParticipating?: boolean
  companyId: number | null
  companyWebsiteUrl: string | null
  companyPublicEmail: string | null
  address: string
}

export type OpportunityFilters = {
  types: OpportunityType[]
  formats: string[]
  tagIds: number[]
  salaryFrom: number | null
  salaryTo: number | null
  statuses: number[]
  verifiedOnly: boolean
}

export const opportunityStatusLabel: Record<number, string> = {
  1: 'Запланировано',
  2: 'На модерации',
  3: 'Активно',
  4: 'Закрыто',
  5: 'Отменено',
  6: 'Отклонено',
  7: 'В архиве',
}

export const typeLabel: Record<OpportunityType, string> = {
  vacancy: 'Вакансия',
  internship: 'Стажировка',
  mentorship: 'Менторство',
  event: 'Мероприятие',
}

export const formatLabel: Record<string, string> = {
  onsite: 'Офис',
  hybrid: 'Гибрид',
  remote: 'Удаленно',
}
